src/mlpack/tests/binarize_test.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * @file binarize_test.cpp
+ * @author Keon Kim
+ *
+ * Test the Binarzie method.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/data/binarize.hpp>
+#include <mlpack/core/math/random.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "old_boost_test_definitions.hpp"
+
+using namespace mlpack;
+using namespace arma;
+using namespace mlpack::data;
+
+BOOST_AUTO_TEST_SUITE(BinarizeTest);
+
+/**
+ * Compare the binarized data with answer.
+ *
+ * @param input The original data set before Binarize.
+ * @param answer The data want to compare with the input.
+ */
+void CheckAnswer(const mat& input,
+                 const umat& answer)
+{
+  for (size_t i = 0; i < input.n_cols; ++i)
+  {
+    const mat& lhsCol = input.col(i);
+    const umat& rhsCol = answer.col(i);
+    for (size_t j = 0; j < lhsCol.n_rows; ++j)
+    {
+      if (std::abs(rhsCol(j)) < 1e-5)
+        BOOST_REQUIRE_SMALL(lhsCol(j), 1e-5);
+      else
+        BOOST_REQUIRE_CLOSE(lhsCol(j), rhsCol(j), 1e-5);
+    }
+  }
+}
+
+BOOST_AUTO_TEST_CASE(BinarizeThreshold)
+{
+  mat input(10, 10, fill::randu); // fill input with randome Number
+  mat constMat(10, 10);
+  math::RandomSeed((size_t) std::time(NULL));
```

Comment:
We should avoid setting the random seed in the tests, this can make specific test errors really hard to reproduce.  What I like to do is set the random seed like you did here and run like 1000 tests on my local machine to make sure it works, then remove the line that sets the seed.